### PR TITLE
Fixing distinct use case

### PR DESCRIPTION
**Pull request by morrowgi**

### CODE REVIEW

File: src/main/java/com/aol/cyclops2/internal/stream/spliterators/DistinctKeySpliterator.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public void forEachRemaining(Consumer<? super T> action) {
         final Consumer<? super IN> toUse = apply(action);
         source.forEachRemaining(e->{
 
-            if(!values.add(keyExtractor.apply(e))){
+            if(values.add(keyExtractor.apply(e))){
```
Comment (davidartplus) [inline src/main/java/com/aol/cyclops2/internal/stream/spliterators/DistinctKeySpliterator.java]: How does this change still comply with getting distinct values for lists with duplicates?

Comment (morrowgi) [inline src/main/java/com/aol/cyclops2/internal/stream/spliterators/DistinctKeySpliterator.java]: So what was happening here was like so, as it was a negative condition it was adding the item if it was returning false. So the following combination would end up getting 2 values added as being distinct:

["java","java","c","c"]

So first "java" would enter condition if(!values.add(keyExtractor.apply(e))) and not get added, second "java" would get added and the same case with "c".

So basically if you just have 1 instance of the element in the list it never gets added, If you have 2 or more they will get added, If you have 3 elements, for example ["Java","Java","Java"]  you would end up with 2 duplicates

File: src/test/java/cyclops/streams/push/reactivestreamspath/ReactiveJDKStreamRSTest.java [hunk 2]
```diff
@@ -78,7 +80,16 @@ public void testDistinct(){
...
 		assertThat(of(1,1,1,2,1).distinct().to(Streamable::fromStream).collect(Collectors.toList()),hasItem(2));
 	}
-	
+
+	@Test
+	public void testDistinctReactiveSeq(){
+		List<String> names = Arrays.asList("Java", "C");
+		ListX<String> d = ReactiveSeq.fromIterable(names).distinct(n -> n + ":" + n).toListX();
+		assertThat(d.size(),is(2));
+	}
```
Comment (morrowgi) [inline src/test/java/cyclops/streams/push/reactivestreamspath/ReactiveJDKStreamRSTest.java]: I will add in some more tests around this also

Comment (davidartplus) [inline src/test/java/cyclops/streams/push/reactivestreamspath/ReactiveJDKStreamRSTest.java]: Maybe just the case of having one value and three equal values

Comment (morrowgi) [inline src/test/java/cyclops/streams/push/reactivestreamspath/ReactiveJDKStreamRSTest.java]: Added tests

**Comment by morrowgi** [comment]
Good point removed the duplicate test

**Pull request merged**